### PR TITLE
Refactor OrderPreferencesController to use annotations

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Refactor Controller to use latest conventions about annotations. Also protect page: now in order to see it you need READ rights;
| Type?         | improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  |  Page "Configure > Shop Parameters > Order Settings" should behave like before. However **especially access rules** are updated: you need READ rights to this page to see it.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/12094)

### CODE REVIEW

**Comment by matks** [comment]
@eternoendless @PierreRambaud I have implemented here the rule "if you have WRITE rights, you have automatically READ rights too". Is that OK ?

**Review by mickaelandrieu — DISMISSED** [review]

File: src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/OrderPreferencesController.php [hunk 1]
```diff
@@ -27,69 +27,51 @@
...
             'help_link' => $this->generateSidebarLink($legacyController),
             'form' => $form->createView(),
             'isMultishippingEnabled' => $this->configuration->getBoolean('PS_ALLOW_MULTISHIPPING'),
             'isAtcpShipWrapEnabled' => $this->configuration->getBoolean('PS_ATCP_SHIPWRAP'),
-        ];
+        ]);
     }
 
     /**
-     * Handle order settings form submit.
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/OrderPreferencesController.php:67]: why did you remove the PHPDoc? I'm curious :)

Comment (matks) [reply src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/OrderPreferencesController.php:67]: Because the controller has only 2 actions: `indexAction` and `processAction`. For me, it's obvious that the 1st handles GET calls and the other one handles form submissions, no need to add it in the phpdoc.

It's like
```
<?php

/**
 * Cat object
 */
class Cat
{
}
```

**Comment by matks** [comment]
PR updated to follow new access rules

**Review by Quetzacoalt91 — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thanks @matks
